### PR TITLE
Re-design CardComponent. Created CardForm, CardSpinner.

### PR DESCRIPTION
**Pull request by ST4R20RD**
- ✔ CardForm:

BEFORE/NOW:
AnimationAnimation

> (CardForm) All tasks Done: 
> - [x] Add/Upload Image Feature
> - [x] Ability to remove one from Selected Participants
> - [x] When User is added, removed from list
> - [x]  Integration with Data Endpoints and new api-hooks for getParticipants

- ✔ Complete Re-Design of Card Component now looking like this: 
image

- ✔CardSpinner:

Animation

### CODE REVIEW

**Review by GAllen98 — APPROVED** [review]

**Review by GAllen98 — CHANGES_REQUESTED** [review]
CardForm incomplete

File: src/components/CardForm/index.tsx [hunk 1]
```diff
@@ -1,3 +1,218 @@
...
+  let date = newDate.getDate();
+  let month = newDate.getMonth() + 1;
+  let year = newDate.getFullYear();
+
+  return `${hour}:${minutes} ${year}${separator}${
+    month < 10 ? `0${month}` : `${month}`
+  }${separator}${date}`;
+}
+
+const mockLocations = [
```
Comment (GAllen98) [inline src/components/CardForm/index.tsx:22]: acho q os locations n vao precisar de id

File: src/components/CardForm/index.tsx [hunk 2]
```diff
@@ -1,3 +1,218 @@
...
+  { id: 1, name: "Group1" },
+  { id: 2, name: "Group2" },
+  { id: 3, name: "Group3" },
+];
+const mockParticipants = [
+  { id: 1, name: "User1" },
+  { id: 2, name: "User2" },
+  { id: 3, name: "User3" },
+];
+const mockCategories = [
```
Comment (GAllen98) [inline src/components/CardForm/index.tsx:37]: same here, as categories podem ser só strings

**Review by GAllen98 — APPROVED** [review]

File: src/components/CardForm/index.tsx [hunk 3]
```diff
@@ -1,3 +1,218 @@
...
+      { date },
+      { duration },
+      { description },
+      { location },
+      { category },
+      { group },
+      { participantsIds }
+    );
+  }, [date, duration, description, location, category, group, participantsIds]);
+
```
Comment (SurgeCode) [inline src/components/CardForm/index.tsx:89]: Remove logging from final pr

File: src/components/CardForm/FormBtnDrop.tsx [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+interface FormBtnDropProps {
+  label: string;
+  list: ItemType[];
+  multiple: boolean;
+  setValue?: React.Dispatch<React.SetStateAction<ItemType>>;
+  setMultiValue?: React.Dispatch<React.SetStateAction<ItemType[]>>;
+  icon?: string;
+  setIsGroupSelected?: React.Dispatch<React.SetStateAction<boolean>>;
+}
```
Comment (SurgeCode) [inline src/components/CardForm/FormBtnDrop.tsx:15]: possibly move types into separate file inside the folder

File: src/components/CardForm/index.tsx [hunk 5]
```diff
@@ -1,3 +1,218 @@
...
+  const [isShared, setIsShared] = useState<boolean>(false);
+
+  const date = getCurrentDate();
+  const [duration, setDuration] = useState<any>();
+  const [description, setDescription] = useState<any>();
+  const [location, setLocation] = useState<any>();
+  const [category, setCategory] = useState<any>();
+  const [group, setGroup] = useState<any>();
+  const [participants, setParticipants] = useState<any[]>([]);
+
```
Comment (SurgeCode) [inline src/components/CardForm/index.tsx:58]: any

**Comment by ST4R20RD** [comment]
Description of Commit AB#{ce8fd83e2549e2cf72cce4adaec3a34e40098c6c} Changes: 
App index
- Created Route Page => Create

Pages
- Added Create Page

CardForm
- Adressed the "any" types to correct types
- Removed anything Groups for POC
- Prepared Button before introduction of "Add Image" feature

Layout
- Handled hidden of Create button when on Create Page

FormButton
- Change Options list from Dropdown to Popup
- Added the ability to Search Users from Popup List
- Added the ability to remove individual selected Users

Modal
- Added the ability to close when click outside

api-hooks
- Added useGetSearchUsers

File: src/components/CardForm/index.tsx (hunk 5 above)
Comment (ST4R20RD) [reply src/components/CardForm/index.tsx]: Adressed https://github.com/ST4R20RD/24Grind-frontend/pull/8#issuecomment-1262795747

File: src/components/CardForm/index.tsx (hunk 2 above)
Comment (ST4R20RD) [reply src/components/CardForm/index.tsx]: The problem is that I need the lists to be in the format { id: number, name: string } to work on the FormBtnDrop component. For the MVP I will try to work on a better solution

File: src/components/CardForm/index.tsx (hunk 1 above)
Comment (ST4R20RD) [reply src/components/CardForm/index.tsx]: Adressed https://github.com/ST4R20RD/24Grind-frontend/pull/8#issuecomment-1262795747

**Pull request merged**